recipes-extended/realsense-camera/realsense-camera/0001-librealsense-Fix-link-when-the-system-library-is-use.patch

```diff
@@ -0,0 +1,30 @@
+From fb74e5eec2368f20c4b43cf9224519b29aee35ad Mon Sep 17 00:00:00 2001
+From: Murilo Belluzzo <murilo.belluzzo@intel.com>
+Date: Wed, 5 Apr 2017 17:06:44 -0700
+Subject: [PATCH 3/3] librealsense: Fix link when the system library is used
+
+Otherwise librealsense won't be listed as a dependency for the nodelet
+that will fail to load due to undefined symbols.
+
+Signed-off-by: Murilo Belluzzo <murilo.belluzzo@intel.com>
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index a1ec2f6..926992a 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -106,6 +106,12 @@ add_library(${PROJECT_NAME}_nodelet src/base_nodelet.cpp src/r200_nodelet.cpp sr
+ target_link_libraries(${PROJECT_NAME}_nodelet
+   ${catkin_LIBRARIES}
+ )
++if(USE_SYSTEM_LIBREALSENSE)
++  target_link_libraries(${PROJECT_NAME}_nodelet
++    ${realsense_LIBRARY}
```

Comment:
Ok then, I see it's a bit too late to fix the patch.

But if you care about the various custom Findrealsense.cmake in the wild then a better solution would be to instruct this CMakeList.txt to ignore all these modules with
```
find_package(realsense REQUIRED NO_MODULE)
```
The librealsense recipe in all branches from fido to morty already provides proper realsenseConfig.cmake which I created exactly to encourage adoption of good cmake practices among librealsense users.

What other uses are you talking about?
The doc page for `find_package()` doesn't state that it's supposed to set anything like that. It's up to good will of a module's author to conform the convention and to set the variable. In fact the manual page [1] suggests that Xxx_LIBRARY "should not usually be used in CMakeFiles.txt". In addition to that I find imported targets (sorry, i mistakenly used the word "alias" for them) more readable than variables not set in the file where they are used. 

For more info on good practices I'd recommend reading [2] and [3].

[1] https://cmake.org/cmake/help/v3.0/manual/cmake-developer.7.html#modules
[2] https://www.slideshare.net/DanielPfeifer1/cmake-48475415
[3] https://archive.fosdem.org/2013/schedule/event/moderncmake/attachments/slides/258/export/events/attachments/moderncmake/slides/258/cmake_fosdem_2013.pdf